src/World.h

```diff
@@ -554,7 +575,37 @@ class cWorld  // tolua_export
 	bool GetBlocks(sSetBlockVector & a_Blocks, bool a_ContinueOnFailure);
 
 	// tolua_begin
-	bool DigBlock   (int a_X, int a_Y, int a_Z);
+
+	/** Replaces the specified block with air, and calls the apropriate block handlers (OnBreaking(), OnBroken()).
+	Wakes up the simulators.
+	Doesn't produce pickups, use DropBlockAsPickups() for that instead.
+	Returns true on success, false if the chunk is not loaded. */
+	bool DigBlock(Vector3i a_BlockPos);
+
+	/** OBSOLETE, use the Vector3-based overload instead.
+	Replaces the specified block with air, and calls the apropriate block handlers (OnBreaking(), OnBroken()).
+	Wakes up the simulators.
+	Doesn't produce pickups, use DropBlockAsPickups() for that instead.
+	Returns true on success, false if the chunk is not loaded. */
+	bool DigBlock(int a_X, int a_Y, int a_Z)
+	{
+		return DigBlock({a_X, a_Y, a_Z});
+	}
+
+	/** Digs the specified block, and spawns the appropriate pickups for it.
```

Comment:
What happens with blocktypes without an appropriate pickup such as air?

Reply:
Good catch with air, it needs a special handler not to drop anything. Although this doesn't get called for air because the client won't break air blocks, this function needs to be safe for it, too.

Anyway, this function spawns anything that the block handler's `ConvertToPickups` returns, for some blocks it's already empty, so no pickups are spawned.

Reply:
Fixed by adding a new mixin for blocks with no drops.